conda_forge_tick/auto_tick.py

```diff
@@ -0,0 +1,930 @@
+        if not any([blas_c]):
+            pluck(total_graph, node)
+
+    # post plucking we can have several strange cases, lets remove all selfloops
+    total_graph.remove_edges_from(nx.selfloop_edges(total_graph))
+
+    top_level = {
+        node for node in total_graph if not list(total_graph.predecessors(node))
+    }
+    cycles = list(nx.simple_cycles(total_graph))
+
+    migrators.append(
+        BlasRebuild(
+            graph=total_graph,
+            pr_limit=5,
+            name="blas-2.0",
+            top_level=top_level,
+            cycles=cycles,
+            obj_version=0,
+        ),
+    )
+
+
+def add_arch_migrate(migrators: MutableSequence[Migrator], gx: nx.Graph):
+    """Adds rebuild migrators.
+
+    Parameters
+    ----------
+    migrators : list of Migrator
+        The list of migrators to run.
+
+    """
+    total_graph = copy.deepcopy(gx)
+
+    for node, node_attrs in gx.nodes.items():
+        attrs = node_attrs["payload"]
+        meta_yaml = attrs.get("meta_yaml", {}) or {}
+        # no need to consider noarch packages for this rebuild
+        noarch = meta_yaml.get("build", {}).get("noarch")
+        if noarch:
+            pluck(total_graph, node)
+        # since we aren't building the compilers themselves, remove
+        if node.endswith("_compiler_stub"):
+            pluck(total_graph, node)
+
+    # post plucking we can have several strange cases, lets remove all selfloops
+    total_graph.remove_edges_from(nx.selfloop_edges(total_graph))
+
+    top_level = {
+        node for node in total_graph if not set(total_graph.predecessors(node))
+    }
+    cycles = list(nx.simple_cycles(total_graph))
+    # print('cycles are here:', cycles)
+
+    migrators.append(
+        ArchRebuild(
+            graph=total_graph,
+            pr_limit=5,
+            name="aarch64 and ppc64le addition",
+            top_level=top_level,
+            cycles=cycles,
+        ),
+    )
+
+
+def add_rebuild_migration_yaml(
+    migrators: MutableSequence[Migrator],
+    gx: nx.DiGraph,
+    package_names: Sequence[str],
+    migration_yaml: str,
+    config={},
+    migration_name="",
+    pr_limit=50,
+):
+    """Adds rebuild migrator.
+
+    Parameters
+    ----------
+    migrators : list of Migrator
+        The list of migrators to run.
+    gx : networkx.DiGraph
+        The feedstock graph
+    package_names : list of str
+        The package who's pin was moved
+    migration_yaml : str
+        The raw yaml for the migration variant dict
+    config: dict
+        The __migrator contents of the migration
+    migration_name: str
+        Name of the migration
+    pr_limit : int, optional
+        The number of PRs per hour, defaults to 5
+    """
+
+    total_graph = copy.deepcopy(gx)
+
+    for node, node_attrs in gx.nodes.items():
+        attrs = node_attrs["payload"]
+        meta_yaml = attrs.get("meta_yaml", {}) or {}
+        if "strong" in meta_yaml.get("build", {}) or any(
```

Comment:
```suggestion
        # TODO: fix this, since it doesn't fully apply the strong constraints
        if "strong" in meta_yaml.get("build", {}) or any(
```